### ISSUE TITLE
The cursor sometimes appears on the wrong line.

### ISSUE THREAD

**Issue by sgrwfwn** [vim/vim#12658]
### Steps to reproduce

The cursor sometimes appears on the wrong line.

I have set “set sms” command. Occasionally, the cursor position is incorrect while using it, and I’m not sure if it’s related to this setting. Please refer to the video below for a specific example:

### Expected behaviour

I hope it can be used normally

### Version of Vim

v9.0.1672

### Environment

Win11

**Comment by chrisbra** [comment]
Note: I redacted the issue, as the video seems to contain a bit too political content. 

@sgrwfwn if you can reproduce it using a vim script testcase, please share here. Videos are not always the best for viewers to understand what's going on. Thanks.

**Comment by sgrwfwn** [comment]
This issue has only occurred a few times, and I am currently not sure why it was triggered. If there are any new developments, I will promptly provide feedback to you.





------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "vim/vim"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年7月12日(星期三) 下午4:10
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [vim/vim] The cursor sometimes appears on the wrong line. (Issue #12658)





 
Note: I redacted the issue, as the video seems to contain a bit too political content.
 
@sgrwfwn if you can reproduce it using a vim script testcase, please share here. Videos are not always the best for viewers to understand what's going on. Thanks.

**Comment by sgrwfwn** [comment]
The cursor has appeared on the wrong line again, and I have been using it normally. It was just a simple editing of the txt file, and I don't know what triggered it.


https://github.com/vim/vim/assets/54033497/1d934102-70ab-4d67-994a-c511a04d8524


https://github.com/vim/vim/assets/54033497/51de5012-b31d

Uploading bandicam 2023-07-22 18-43-50-276.mp4…

-4ecd-a498-46c46cf97c5b

**Comment by sgrwfwn** [comment]
https://github.com/vim/vim/assets/54033497/a1a6fb99-16e1-469f-aa7a-b82ac957d992

**Comment by sgrwfwn** [comment]
@chrisbra

**Comment by chrisbra** [comment]
yes what? Please provide a reproducible example. We need to know what keys were pressed, and I cannot guess from your video.
